erizo/src/erizo/lib/TokenBucket.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright (c) 2016 Erik Rigtorp <erik@rigtorp.se>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+#include "TokenBucket.h"
+
+#include <cassert>
+#include <iostream>
+
+namespace erizo {
+
+TokenBucket::TokenBucket(std::shared_ptr<erizo::Clock> the_clock)
+  : time_{0},
+    time_per_token_{0},
+    time_per_burst_{0},
+    clock_{the_clock} {}
+
+TokenBucket::TokenBucket(const uint64_t rate, const uint64_t burst_size,
+            std::shared_ptr<erizo::Clock> the_clock)
+  : time_ {0},
+    time_per_token_{1000000 / rate},
+    time_per_burst_{burst_size * time_per_token_},
+    clock_{the_clock} {
+  assert(rate > 0 && "Rate should be greater than 0");
```

Comment:
Not convinced about using `assert` mostly because we're not using it in general.
We should discuss if we want to use it in general